### PR TITLE
[develop] Trello-VOXOpcNb: Transactions API

### PR DESCRIPTION
**Pull request by AndreySurzhan**
[Trello-VOXOpcNb](https://trello.com/c/VOXOpcNb): Transactions API

- [X] possible to add new Transaction 

Request
```
{
	"note": "New Transaction - ${new Date().getTime()}",
	"value" : 667,
	"type": "income",
	"accountSource": "${accountSrouce}",
	"accountDestination": "${accountDestination}",
	"category": "${category}"
}
```

Response
```
{
   "__v": 0,
   "projectId": "5a3b1ba7ee10f07e84fa0ee4",
   "type": "income",
   "value": 667,
   "note": "New Transaction - 1513824033555",
   "category": "5a39df1f1c7b0a7c036c1f78",
   "accountSource": "5a3b1e90e67ea07ee4a1fb0a",
   "accountDestination": "5a3b1e90e67ea07ee4a1fb0a",
   "created": "2017-12-21T02:40:33.578Z",
   "createdBy": "5a371ad09eabb8740cdef3e4",
   "modifiedBy": "5a371ad09eabb8740cdef3e4",
   "_id": "5a3b1f21e67ea07ee4a1fb0e",
   "modified": "2017-12-21T02:40:33.578Z"
}
```

- [x] possible to update transaction
- [x] possible to get all transactions by project id
- [x] possible to delete transaction
- [x] API autotests for transactions (basic)
 
NOTE: Project account values don't get updated automatically on adding new transaction
NOTE: Accounts and categories don't get populated in the response since they are sub docs of the project

### CODE REVIEW

File: src/controllers/account.js [hunk 1]
```diff
@@ -276,6 +236,162 @@ module.exports = {
...
+                })
+                .catch((error) => {
+                    deferred.reject(error);
+                });
+
+            return deferred.promise;
+        }
+
+        function updateAccById () {
+            let deferred = Q.defer();
```
Comment (wildeastengineer) [inline src/controllers/account.js]: why let?

File: src/controllers/project.js [hunk 2]
```diff
@@ -414,5 +415,30 @@ module.exports = {
      */
     deleteAccount(data) {
         return AccountController.deleteAccount(data);
+    },
+
```
Comment (wildeastengineer) [inline src/controllers/project.js]: extra line

File: src/controllers/transaction.js [hunk 3]
```diff
@@ -1,12 +1,289 @@
...
+     * @param {?(ObjectId|String)} data.transaction.category
+     * @param {?(ObjectId|String)} data.transaction.accountSource
+     * @param {?(ObjectId|String)} data.transaction.accountDestination
+     *
+     * @returns {Promise<models/AccountSchema|Error>}
+     */
+    post(data) {
+        //find project to figure out whether use is able to create transactions
+        let deferred = Q.defer();
+        let that = this;
```
Comment (wildeastengineer) [inline src/controllers/transaction.js]: do we support arrow-functions?

**Comment by AndreySurzhan** [comment]
@wildeastengineer please review

**Pull request merged**